lib/engine/game/g_1840/step/reassign_trains.rb

```diff
@@ -69,7 +69,7 @@ def process_reassign_trains(action)
                     "#{invalid_tram_corp.full_name} cannot be assigned more than one train"
             end
 
-            if entity.trains.size.positive? && !@game.tram_owned_by_corporation[entity].empty? { |item| item.trains.size.zero? }
+            if entity.trains.size.positive? && !@game.tram_owned_by_corporation[entity].empty? { |item| item.trains.empty? }
```

Comment:
entity.trains.empty?

why size.positive?

Reply:
needs to be >0 . what do you prefer?

Reply:
i could just make !empty